UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaDataLayer.cs

```diff
@@ -1,22 +1,39 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 using Systems.Atmospherics;
 using Chemistry;
 using Chemistry.Components;
 using HealthV2;
 using Items;
+using Messages.Server;
+using Mirror;
 using Newtonsoft.Json;
 using UnityEngine;
 using Objects.Construction;
 using TileManagement;
+using Random = UnityEngine.Random;
 
 /// <summary>
 /// Holds and provides functionality for all the MetaDataTiles for a given matrix.
 /// </summary>
 public class MetaDataLayer : MonoBehaviour
 {
-	private SerializableDictionary<Vector3Int, MetaDataNode> nodes = new SerializableDictionary<Vector3Int, MetaDataNode>();
 
+	//Easiest to synchronise?
+	//Attribute
+	//What needs to be done
+
+	//set current state on clients
+	//Only send the different node
+
+
+	private Dictionary<Vector3Int, MetaDataNode> nodes = new Dictionary<Vector3Int, MetaDataNode>();
+
+
+	public Dictionary<Vector3Int, MetaDataNode> ChangedNodes = new Dictionary<Vector3Int, MetaDataNode>(); //Used for networking
+
+	public List<MetaDataNode> nodesToUpdate = new List<MetaDataNode>();
```

Comment:
more efficient have a list

Reply:
you also need to be separate so, changes per frame can be worked out

Reply:
I'm skeptical about it being faster (unless it's queried more than inserted into) but I'm going to assume you've profiled it.

Reply:
list is just there because you're inserting changes, you never query it

Reply:
Looking at the code again I understand it now. `ChangedNodes` contains the changes for the entire lifetime of the map, `nodesToUpdate` only the changes per frame. I think a comment explaining this would be great.